Cargo.toml

```diff
@@ -16,8 +16,8 @@ exclude = [
   "**/ion-tests/iontestdata/**",
   "*.pdf"
 ]
-version = "0.7.0"
-edition = "2018"
+version = "0.8.0"
```

Comment:
`ion-hash`, `ion-c-sys`, and `ion-c-sys-macros` have not been modified since their last change. The edition bump doesn't affect their APIs, so I've bumped their patch version (The `z` in `x.y.z`).

In contrast, `ion-rs` has landed some substantive changes since `v0.7.0` (for example: the `SystemReader` in #344), so it's getting a minor version bump to `v0.8.0`.